kuma/core/jobs.py

```diff
@@ -37,6 +39,46 @@ def hash(self, value):
         return hashlib.md5(six.b(':').join(value)).hexdigest()
 
 
+class GenerationJob(KumaJob):
+    generation_age = 60 * 60 * 24 * 365
+
+    def __init__(self, *args, **kwargs):
+        self.generation_args = kwargs.pop('generation_args', [])
+        super(KumaJob, self).__init__(*args, **kwargs)
+
+    @cached_property
+    def generation_key(self):
+        generation_args = ':'.join([str(key) for key in self.generation_args])
+        return '%s:%s:generation' % (self.class_path, generation_args)
+
+    def key(self, *args, **kwargs):
+        key = super(GenerationJob, self).key(*args, **kwargs)
+        return '%s@%s' % (key, self.generation())
+
+    def generation(self):
```

Comment:
Not really a subset, but an addition, which is why I wanted to put this into a subclass. This still uses the main features of cacheback but simply adds another invalidation scheme via a cache key generation additionally to the default one-cache-key-per-job-parameters technique.